code-challenges.test.js

```diff
@@ -15,46 +17,152 @@
 
 
 // b) Create the function that makes the test pass.
 
+//Take input of given array into a const
+//Create a filter that removes all types of data except for numbers
+//Order the filtered numbers from least to greatest
+//Return new array with only numbers
+
+const sorted = (val) => {
+    if(typeOf(val) === 'number'){
+        let newArray = sorted.slice().sort((a,b)=>a-b)
+            return newArray
+    
+    }
+    }
+    
+//Test FAIL. Received undefined / typeOf is not defined
+   
+//Notes on process: Having difficulty with this one. I believe that the logic I am following in my code is correct, but I am not sure about how to write it.
+
+
 
 
 // --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-var numbersToAdd1 = [2, 4, 45, 9]
-// Excpected output: [2, 6, 51, 60]
+describe("Accumulating Sum", () => {
 
-var numbersToAdd2 = [0, 7, -8, 12]
-// Expected output: [0, 7, -1, 11]
+    // a test/it method, nested within the describe block, that in plain words, describes that the function does.
+    it("Returns array with accumulating sum", () => {
+  
+      //an expect method, nested within the test block, calling on the hello() function, followed by the .toEqual() matcher that checks the expected output of the function return.
+      expect(numbers([2, 4, 45, 9])).toEqual(([2, 6, 51, 60]))
+      expect(numbers([0, 7, -8, 12])).toEqual(([0, 7, -1, 11]))
+      expect(numbers([])).toEqual(([]))
+    })
+  })
 
-var numbersToAdd3 = []
-// Expected output: []
+//   Test FAIL. ReferenceError: numbers is not defined
 
+// var numbersToAdd1 = [2, 4, 45, 9]
+// // Excpected output: [2, 6, 51, 60]
 
+// var numbersToAdd2 = [0, 7, -8, 12]
+// // Expected output: [0, 7, -1, 11]
 
-// b) Create the function that makes the test pass.
+// var numbersToAdd3 = []
+// // Expected output: []
+
+
+
+// // b) Create the function that makes the test pass.
+
+//Create a an anonymous function named numbers that takes in an array
+//use .map to get the sum of each value within the given arrays
+//return the array after the addition is complete
+
+const numbers = (arr)=> {
+    arr.map((sum => value => sum += value)(0));
+return arr
+    
+}
+
+//Test FAIL. expect(received).toEqual(expected) // deep equality
+
+    // - Expected  - 3
+    // + Received  + 3
+
+    //   Array [
+    //     2,
+    // -   6,
+    // -   51,
+    // -   60,
+    // +   4,
+    // +   45,
+    // +   9,
+
+
+//Notes on process: Again my logic seems to be correct and it seems like half of the answers are correct in the jest test. Maybe my test isn't setup correctly.
+//I found the way to .map this on codegrepper. This was a difficult week! I spent lots of time just trying to figure out my syntax issues, but I still have difficulty with researching the exact problem.
```

Comment:
Yea, this problem is helped by writing out each logical step as a variable inside of a function. That said the only thing I can see that might fix this code has to do with the parenthese.